ts/components/leftPane/LeftPaneSearchHelper.tsx

```diff
@@ -298,10 +306,29 @@ export class LeftPaneSearchHelper extends LeftPaneHelper<LeftPaneSearchPropsType
     );
   }
 
-  // This is currently unimplemented. See DESKTOP-1170.
   getConversationAndMessageAtIndex(
     _conversationIndex: number
   ): undefined | { conversationId: string; messageId?: string } {
+    if (_conversationIndex < 0) {
+      return undefined;
+    }
+    let pointer = _conversationIndex;
+    for (const list of this.allResults()) {
+      if (list.isLoading) {
+        continue;
+      }
+      if (pointer < list.results.length) {
+        const result = list.results[pointer];
+        return 'conversationId' in result
```

Comment:
might be fine with how this is already structured but this is a little brittle, I could imagine we add different types of results later on that aren't messages but have a `conversationId` and this check has this assumption. I would certainly prefer a `type: 'message'` here.

Reply:
You are right this check is not futureproof. I need your help here: where should this `type: 'message'` field come from? Message results are passed via constructor: do I modify results there?

Reply:
Fixed, now code checks if type is 'ingoing'|'outgoing' to find out if result is a message